addons/web_editor/static/src/js/wysiwyg/widgets/link_popover_widget.js

```diff
@@ -168,7 +168,10 @@ const LinkPopoverWidget = Widget.extend({
      * @param {Event} ev
      */
     _onEditLinkClick(ev) {
-        $('#toolbar #create-link').click();
+        $('#wrapwrap').data('wysiwyg').toggleLinkTools({
```

Comment:
web_editor (and even website editor) should not suppose `$('#wrapwrap')` exists (note: the link popover is not present in mass mailing?). And even if it could, accessing the editor that way is not good but I guess it's ok if the editor team put that data wysiwyg on it... :shrug: Can't this be a trigger_up which will reach that wysiwyg instance instead? :thinking: 

If not, at least put a TODO comment for the day I'll have time to refactor the editor mess :grimacing: Something like "TODO the editor instance should be reached another way" :+1:

Reply:
> If not, at least put a TODO comment 

The TODO is/was already here, in the docstring.

> note: the link popover is not present in mass mailing?

AFAIK no, they have their own popover (basically the toolbar) so they dont want/need this

> And even if it could, accessing the editor that way is not good but I guess it's ok if the editor team put that data wysiwyg on it.

Yup, they even also write themself `$('#wrapwrap').data('wysiwyg')` in this file
https://github.com/odoo-dev/odoo/commit/740168ce8d27da3d6a7156d2d79655a898394923#diff-44e2f4c9175434219f95516b77671359e8299bde247501bcd945c5e68c59c1bbR176

Reply:
@rdeodoo That TODO is not true anymore :thinking: You are now calling business method of the editor and you are not clicking anymore?

Reply:
Changed it
-     * TODO Call business methods once new editor is released instead of click
+     * TODO The editor instance should be reached a proper way